### PR TITLE
Add otel tracing

### LINKED ISSUE #2264: Implement native Opentelemetry tracing in the project
Similar to so many other projects: https://github.com/kyverno/kyverno/blob/9b41e2e0174c10b3548a5364e205911d6a0047c7/pkg/tracing/tracing.go

Improve observability of the project by implementing OpenTelemetry tracing.

### PR DESCRIPTION
**Pull request by yuriolisa**
## Problem Statement
Implementing a trace package using the OpenTelemetry SDK.

What is the problem you're trying to solve?
Lack of ESO traces.

## Related Issue

Resolves #2264

## Proposed Changes

Implemented trace package using the Opentelemetry native sdk and created the required stuff to set environment variables to point out the related variables.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by moolen** [comment]
@yuriolisa there are no changes yet, this is a WIP, right? Going to put it into a PR draft :)

**Comment by moolen** [comment]
@yuriolisa any update on this PR? :pray:

**Comment by yuriolisa** [comment]
@moolen, @knelasevero sorry for this taking so long. 

I have added the minimal parameters to get a functional OpenTelemetry instrumentation, other than that, as a test, I started instrumenting the GetSecret function in Azure provider. 

I appreciate if you could have a look at it.

**Comment by knelasevero** [comment]
Hey @yuriolisa do you think you can add some docs together with this addition?

**Comment by yuriolisa** [comment]
I'll work on it this week.

File: Tiltfile [hunk 1]
```diff
@@ -80,7 +80,7 @@ if settings.get('debug').get('enabled'):
 
 
 docker_build_with_restart(
-    'ghcr.io/external-secrets/external-secrets',
+    'localhost:5000/external-secrets',
```
Comment (Skarlso) [inline Tiltfile:83]: This should stay ghcr, since that is how Tilt will find the right container to replace. :)

**Comment by Skarlso** [comment]
@yuriolisa any plans on continuing this?

**Comment by jensholdgaard** [comment]
i will pick up the mantle here perhaps soon

**Pull request closed without merging**